os/services/shell/shell-commands.c

```diff
@@ -734,7 +735,39 @@ shell_commands_init(void)
                                     echo_reply_handler);
 }
 /*---------------------------------------------------------------------------*/
-struct shell_command_t shell_commands[] = {
+void
+shell_command_set_register(struct shell_command_set_t *set)
+{
+  set->next = shell_command_sets_list;
+  shell_command_sets_list = set;
```

Comment:
Please use the list library `os/lib/list.h`, gives you off-the-shelf singly linked list so you don't have to re-implement.

Reply:
Ah cool, thanks. Will update the PR.